connaisseur/validate.py

```diff
@@ -98,9 +98,13 @@ def process_chain_of_trust(
                 host, image, TUFRole(delegation)
             )
 
-    # validate all trust data's signatures, expiry dates and hashes
+    # validate all trust data's signatures, expiry dates and hashes.
+    # when delegation are added to the repository, but weren't yet used for signing, the
+    # delegation files don't exist yet and are `None`. in this case they can't be
+    # validated and must be skipped.
     for role in trust_data:
-        trust_data[role].validate(key_store)
+        if trust_data[role]:
```

Comment:
`if trust_data[role] is not None:`

Otherwise, it depends on the truthiness of the value itself. It may not be a problem here, but it's good practice to be explicit about `None` (or the absence of things in general).